### PR TITLE
Port the code to the third revision PCB

### PR DESCRIPTION
**Pull request by antoinealb**
Tested on hardware today, this PR brings the code to work on the latest version of the board that was assembled by JLCPCB. I have tested it with another CAN dongle (from the previous iteration) and both `candump` and the UAVCAN GUI tool.

edit: also tested, works well with the raspberrypi

### CODE REVIEW

File: src/bus_power.c [hunk 1]
```diff
@@ -39,10 +40,10 @@ static const ADCConversionGroup adcgrpcfg = {
     ADC_TR(0, 4095), /* TR1     */
     0, /* CCR     */
     {/* SMPR[2] */
-     ADC_SMPR1_SMP_AN3(ADC_SMPR_SMP_601P5),
+     ADC_SMPR1_SMP_AN3(ADC_SMPR_SMP_601P5) | ADC_SMPR1_SMP_AN5(ADC_SMPR_SMP_601P5),
```
Comment (nuft) [inline src/bus_power.c:43]: what does `ADC_SMPR1_SMP_AN5(ADC_SMPR_SMP_601P5)` do?

**Review by nuft — APPROVED** [review]
LGTM. Thanks for fixing the bugs and isolating board specific code.

File: src/board/CAN-USB-dongle/RevA/board.c [hunk 2]
```diff
@@ -130,7 +130,7 @@ void usb_lld_disconnect_bus(void* arg)
     (void)arg;
     palSetPadMode(GPIOA, GPIOA_OTG_FS_DM, PAL_MODE_OUTPUT_OPENDRAIN);
     palSetPadMode(GPIOA, GPIOA_OTG_FS_DP, PAL_MODE_OUTPUT_OPENDRAIN);
-    palClearPort(GPIOA, GPIOA_OTG_FS_DM | GPIOA_OTG_FS_DP);
+    palClearPort(GPIOA, (1 << GPIOA_OTG_FS_DM) | (1 << GPIOA_OTG_FS_DP));
```
[the remark this answers is not in the document]
Comment (antoinealb) [reply src/board/CAN-USB-dongle/RevA/board.c:133]: Yes. But it did not have any consequences, until I changed the pinout for the new PCB.

File: src/bus_power.c (hunk 1 above)
Comment (antoinealb) [reply src/bus_power.c:43]: It sets up the analog channel 5, which is used for vbus sensing on the rev C. If that channel is unused, then it does not have any impact.

**Pull request merged**